commit cca86038eed2ba8b3657280a2b5ac406aac6f870 (refs/changes/37/5537/3)
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-02-26T12:48:03+00:00 (7 months ago)
    
    TRITON-1130 convert sdc-rabbitmq to engbld framework
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/.gitignore b/.gitignore
index 5050e8a..6d0b477 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1 +1,3 @@
+/make_stamps
 /rabbitmq-pkg-*.tar.bz2
+bits
diff --git a/.gitmodules b/.gitmodules
index 48ab5ca..14090f5 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,6 @@
 [submodule "deps/sdc-scripts"]
 	path = deps/sdc-scripts
 	url = https://github.com/joyent/sdc-scripts.git
+[submodule "deps/eng"]
+	path = deps/eng
+	url = https://github.com/joyent/eng.git
diff --git a/Makefile b/Makefile
index 7b55ad9..7cc732c 100644
--- a/Makefile
+++ b/Makefile
@@ -5,45 +5,54 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2019, Joyent, Inc.
 #
 
 NAME=rabbitmq
-
-ifeq ($(VERSION), "")
-    @echo "Use gmake"
-endif
-
 TAR = tar
-
-ifeq ($(TIMESTAMP),)
-    TIMESTAMP=$(shell date -u "+%Y%m%dT%H%M%SZ")
+RELEASE_TARBALL=$(NAME)-pkg-$(STAMP).tar.gz
+
+BASE_IMAGE_UUID = fd2cc906-8938-11e3-beab-4359c665ac99
+BUILDIMAGE_NAME = $(NAME)
+BUILDIMAGE_DESC	= SDC RabbitMQ
+BUILDIMAGE_PKGSRC = \
+	perl-5.14.2nb3 \
+	iodbc-3.52.7 \
+	libffi-3.0.9nb1 \
+	python27-2.7.2nb2 \
+	py27-setuptools-0.6c11nb1 \
+	py27-simplejson-2.1.1 \
+	erlang-14.1.4nb1 \
+	rabbitmq-2.7.1
+AGENTS		= amon config registrar
+
+ENGBLD_USE_BUILDIMAGE	= true
+ENGBLD_REQUIRE		:= $(shell git submodule update --init deps/eng)
+include ./deps/eng/tools/mk/Makefile.defs
+TOP ?= $(error Unable to access eng.git submodule Makefiles.)
+
+ifeq ($(shell uname -s),SunOS)
+    include ./deps/eng/tools/mk/Makefile.agent_prebuilt.defs
 endif
 
-RABBITMQ_PUBLISH_VERSION := $(shell git symbolic-ref HEAD | \
-      awk -F / '{print $$3}')-$(TIMESTAMP)-g$(shell \
-                git describe --all --long | awk -F '-g' '{print $$NF}')
-
-RELEASE_TARBALL=rabbitmq-pkg-$(RABBITMQ_PUBLISH_VERSION).tar.bz2
-
 .PHONY: all
-
 all: sdc-scripts
 
-release: $(RELEASE_TARBALL)
+.PHONY: release
+release: all $(RELEASE_TARBALL)
 
 $(RELEASE_TARBALL):
 	TAR=$(TAR) bash package.sh $(RELEASE_TARBALL)
 
 publish:
-	@if [[ -z "$(BITS_DIR)" ]]; then \
-      echo "error: 'BITS_DIR' must be set for 'publish' target"; \
-      exit 1; \
-    fi
-	mkdir -p $(BITS_DIR)/rabbitmq
-	cp $(RELEASE_TARBALL) $(BITS_DIR)/rabbitmq/$(RELEASE_TARBALL)
+	mkdir -p $(ENGBLD_BITS_DIR)/rabbitmq
+	cp $(RELEASE_TARBALL) $(ENGBLD_BITS_DIR)/rabbitmq/$(RELEASE_TARBALL)
 
-clean:
-	rm -fr rabbitmq-pkg-*.tar.bz2
 
 sdc-scripts: deps/sdc-scripts/.git
+
+ifeq ($(shell uname -s),SunOS)
+    include ./deps/eng/tools/mk/Makefile.agent_prebuilt.targ
+endif
+include ./deps/eng/tools/mk/Makefile.deps
+include ./deps/eng/tools/mk/Makefile.targ
diff --git a/deps/eng b/deps/eng
new file mode 160000
index 0000000..7c472bc
--- /dev/null
+++ b/deps/eng
@@ -0,0 +1 @@
+Subproject commit 7c472bc495ba44ac4cdde0af50ff627021942524
diff --git a/package.sh b/package.sh
index 618977c..b866903 100755
--- a/package.sh
+++ b/package.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2019, Joyent, Inc.
 #
 
 set -o xtrace
@@ -38,6 +38,6 @@ mkdir -p ${tmpdir}/root/opt/smartdc/boot
 cp -R ${ROOT}/deps/sdc-scripts/* ${tmpdir}/root/opt/smartdc/boot/
 cp -R ${ROOT}/boot/* ${tmpdir}/root/opt/smartdc/boot/
 
-(cd ${tmpdir}; tar -jcf ${ROOT}/${RELEASE_TARBALL} root site)
+(cd ${tmpdir}; gtar -I pigz -cf ${ROOT}/${RELEASE_TARBALL} root site)
 
 rm -rf ${tmpdir}
